shared/utils/yaml-to-shorthand.py

```diff
@@ -0,0 +1,531 @@
+        root = self.to_xml_element()
+        tree = ET.ElementTree(root)
+        tree.write(file_name)
+
+    def add_value(self, value):
+        if value is None:
+            return
+        self.values[value.id_] = value
+
+    def add_group(self, group):
+        if group is None:
+            return
+        self.groups[group.id_] = group
+
+    def add_rule(self, rule):
+        if rule is None:
+            return
+        self.rules[rule.id_] = rule
+
+    def to_xccdf(self):
+        """We can easily extend this script to generate a valid XCCDF instead
+        of SSG SHORTHAND.
+        """
+        raise NotImplementedError
+
+    def __str__(self):
+        return self.id_
+
+
+class Group(object):
+    """Represents XCCDF Group
+    """
+    def __init__(self, id_):
+        self.id_ = id_
+        self.values = {}
+        self.groups = {}
+        self.rules = {}
+
+    @staticmethod
+    def from_yaml(yaml_file):
+        yaml_contents = open_yaml(yaml_file)
+        if yaml_contents is None:
+            return None
+
+        group_id, _ = os.path.splitext(os.path.basename(yaml_file))
+        group = Group(group_id)
+        group.prodtype = yaml_contents.get("prodtype", "all")
+        group.title = yaml_contents['title']
+        group.description = yaml_contents['description']
+        return group
+
+    def to_xml_element(self):
+        group = ET.Element('Group')
+        group.set('id', self.id_)
+        if self.prodtype != "all":
+            group.set("prodtype", self.prodtype)
+        title = ET.SubElement(group, 'title')
+        title.text = self.title
+        add_sub_element(group, 'description', self.description)
+        for v in self.values.values():
+            group.append(v.to_xml_element())
+        for g in self.groups.values():
+            group.append(g.to_xml_element())
+        for r in self.rules.values():
+            group.append(r.to_xml_element())
+        return group
+
+    def to_file(self, file_name):
+        root = self.to_xml_element()
+        tree = ET.ElementTree(root)
+        tree.write(file_name)
+
+    def add_value(self, value):
+        if value is None:
+            return
+        self.values[value.id_] = value
+
+    def add_group(self, group):
+        if group is None:
+            return
+        self.groups[group.id_] = group
+
+    def add_rule(self, rule):
+        if rule is None:
+            return
+        self.rules[rule.id_] = rule
+
+    def __str__(self):
+        return self.id_
+
+
+class Rule(object):
+    """Represents XCCDF Rule
+    """
+    def __init__(self, id_):
+        self.id_ = id_
+
+    @staticmethod
+    def from_yaml(yaml_file):
+        yaml_contents = open_yaml(yaml_file)
+        if yaml_contents is None:
+            return None
+
+        rule_id, _ = os.path.splitext(os.path.basename(yaml_file))
+        rule = Rule(rule_id)
+        rule.prodtype = yaml_contents.get("prodtype", "all")
+        rule.title = yaml_contents['title']
```

Comment:
An idea for future:

I think it would be useful to check if all required XCCDF tags are present in the YAML and if they are not add a nice error message so that the YAML contributors know where they made a mistake.
eg:
`Error: Missing required 'description' field in ssh_use_root_login.yml.`

Reply:
Thx, this is fixed now. It's not as good as you suggest but it's decent enough. I would need to pass more info to the parser functions to get the actual file. This can be done in the future.